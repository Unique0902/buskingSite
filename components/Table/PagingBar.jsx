import React from 'react';
import { ArrowLeftIcn, ArrowRightIcn } from '../../assets/icon/icon';

export default function PagingBar({
  resultNum,
  pageNum,
  onPagePlus,
  onPageMinus,
}) {
  const btnStyle = ' rounded-full hover:scale-110 px-4 py-2';

  return (
    <div className='flex flex-row items-center justify-center mt-3'>
      <button className={btnStyle} onClick={onPageMinus}>
        <ArrowLeftIcn width={18} height={18} color={'white'} />
      </button>
      <p className='ml-2 font-sans text-xl font-normal text-white'>
        {pageNum} / {parseInt((resultNum - 1) / 6) + 1}
      </p>
      <button className={btnStyle} onClick={onPagePlus}>
        <ArrowRightIcn width={18} height={18} color={'white'} />
      </button>
    </div>
  );
}
