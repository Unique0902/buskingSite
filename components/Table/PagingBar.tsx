import React from 'react';
import { ArrowLeftIcn, ArrowRightIcn } from '../../assets/icon/icon';
type Props = {
  resultNum: number;
  pageNum: number;
  onPagePlus: () => void;
  onPageMinus: () => void;
};
export default function PagingBar({
  resultNum,
  pageNum,
  onPagePlus,
  onPageMinus,
}: Props) {
  const btnStyle = ' rounded-full hover:scale-110 px-4 py-2';

  return (
    <div className='flex flex-row items-center justify-center mt-3'>
      <button className={btnStyle} onClick={onPageMinus}>
        <ArrowLeftIcn width={18} height={18} color={'white'} />
      </button>
      <p className='ml-2 text-xl font-normal text-white'>
        {pageNum} / {Math.floor((resultNum - 1) / 6) + 1}
      </p>
      <button className={btnStyle} onClick={onPagePlus}>
        <ArrowRightIcn width={18} height={18} color={'white'} />
      </button>
    </div>
  );
}
