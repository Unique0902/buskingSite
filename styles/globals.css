@tailwind base;
@tailwind components;
@tailwind utilities;
/* TODO: 글꼴 불러오는 로딩 줄이거나 없애기 */
* {
  box-sizing: border-box;
  font-family: 'Pretendard';
}

@font-face {
  font-family: 'Pretendard';
  font-weight: 45 920;
  font-style: normal;
  font-display: swap;
  src: url('../assets/fonts/PretendardVariable.woff2')
    format('woff2-variations');
}

li {
  list-style: none;
}
body {
  margin: 0;
  @apply text-black dark:text-white dark:bg-slate-600;
}
p,
h1,
h2 {
  margin: 0;
}
ul {
  padding: 0;
}
