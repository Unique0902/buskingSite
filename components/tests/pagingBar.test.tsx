import { describe, expect, it, jest } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import PagingBar from '../Table/PagingBar';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

describe('PagingBar component test', () => {
  it('rendering correctly test', () => {
    const component = renderer.create(
      <PagingBar
        totalPageNum={2}
        pageNum={1}
        onPageMinus={() => {}}
        onPagePlus={() => {}}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('show totalPageNum and nowPageNum correctly', () => {
    render(
      <PagingBar
        totalPageNum={2}
        pageNum={1}
        onPageMinus={() => {}}
        onPagePlus={() => {}}
      />
    );
    expect(screen.queryByText('1')).not.toBeNull();
    expect(screen.queryByText('2')).not.toBeNull();
  });

  it('click each plus, minus btn, and get event onPagePlus and onPageMinus', async () => {
    const handlePagePlus = jest.fn();
    const handlePageMinus = jest.fn();
    render(
      <PagingBar
        totalPageNum={2}
        pageNum={1}
        onPagePlus={handlePagePlus}
        onPageMinus={handlePageMinus}
      />
    );

    const [minusBtn, plusBtn] = screen.queryAllByRole('button');

    await userEvent.click(plusBtn);
    expect(handlePagePlus).toHaveBeenCalled();
    await userEvent.click(minusBtn);
    expect(handlePageMinus).toHaveBeenCalled();
  });
});
