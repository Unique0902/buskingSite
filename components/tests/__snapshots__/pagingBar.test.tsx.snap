// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagingBar component test rendering correctly test 1`] = `
<div
  className="flex flex-row items-center justify-center mt-3"
>
  <button
    className=" rounded-full hover:scale-110 px-4 py-2"
    onClick={[Function]}
  >
    <svg
      fill="white"
      height={18}
      viewBox="0 0 306 306"
      width={18}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M247.35 35.7 211.65 0l-153 153 153 153 35.7-35.7L130.05 153z"
        fillRule="nonzero"
      />
    </svg>
  </button>
  <div
    className="flex flex-row items-center gap-2 text-xl font-normal text-white"
  >
    <p>
      1
    </p>
    <p>
      /
    </p>
    <p>
      2
    </p>
  </div>
  <button
    className=" rounded-full hover:scale-110 px-4 py-2"
    onClick={[Function]}
  >
    <svg
      fill="white"
      height={18}
      viewBox="0 0 306 306"
      width={18}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M58.65 267.75 175.95 153 58.65 35.7 94.35 0l153 153-153 153z"
        fillRule="nonzero"
      />
    </svg>
  </button>
</div>
`;
